frontend/packages/kubevirt-plugin/src/components/modals/scheduling-modals/shared/NodeChecker/node-checker.tsx

```diff
@@ -25,48 +30,67 @@ import './node-checker.scss';
 
 export const NodeChecker: React.FC<NodeCheckerProps> = ({
   qualifiedNodes,
+  qualifiedPerferredNodes,
   wariningTitle,
   warningMessage,
 }) => {
   const size = qualifiedNodes.length;
   const buttonText = pluralize(size, 'Node');
+
+  const isNodePreferred = (node: NodeKind) =>
+    qualifiedPerferredNodes?.some((prefNode) => getName(node) === getName(prefNode));
+
   return (
     <Alert
       className="kv-node-checker"
       variant={size > 0 ? 'success' : 'warning'}
       isInline
       title={
         size > 0
-          ? SCHEDULING_NODES_MATCH_TEXT(size)
+          ? qualifiedPerferredNodes?.length > 0
+            ? SCHEDULING_WITH_PREFERRED_NODES_MATCH_TEXT(size, qualifiedPerferredNodes.length)
+            : SCHEDULING_NODES_MATCH_TEXT(size)
           : wariningTitle || SCHEDULING_NO_NODES_MATCH_TEXT
       }
     >
-      <Popover
-        headerContent={<div>{buttonText} found</div>}
-        position={PopoverPosition.right}
-        className="kv-node-checker__popover"
-        bodyContent={qualifiedNodes.map((node) => (
-          <ExternalLink
-            key={getName(node)}
-            href={resourcePath('Node', getName(node))}
-            text={<ResourceLink linkTo={false} kind="Node" name={getName(node)} />}
-          />
-        ))}
-      >
-        <Button isInline isDisabled={size === 0} variant="link">
-          <Text component={TextVariants.h4}>
-            {size > 0
-              ? SCHEDULING_NODES_MATCH_BUTTON_TEXT(size)
-              : warningMessage || SCHEDULING_NO_NODES_MATCH_BUTTON_TEXT}
-          </Text>
-        </Button>
-      </Popover>
+      <Stack>
+        <StackItem>
+          <Popover
+            headerContent={<div>{buttonText} found</div>}
+            position={PopoverPosition.right}
+            className="kv-node-checker__popover"
+            bodyContent={qualifiedNodes
+              .sort((a) => (isNodePreferred(a) ? -1 : 1))
```

Comment:
we could also sort it by the weight or by the name

Reply:
weight is an affinity field, not accessible from the generic node checker.

Reply:
ok